### PR TITLE
[CI-CD] impl

### LINKED ISSUE #43: [CI/CD] Set up CI/CD pipeline with GitHub Actions

### PR DESCRIPTION
**Pull request by wweqg**
Close #43

### CODE REVIEW

**Comment by wweqg** [comment]
The env variables should be included in a separate config file instead, I will work on it after ensuring everything works correctly.

**Comment by wweqg** [comment]
> The env variables should be included in a separate config file instead, I will work on it after ensuring everything works correctly.

Updated

**Comment by wweqg** [comment]
> > The env variables should be included in a separate config file instead, I will work on it after ensuring everything works correctly.
> 
> Updated

Doesn't work. Roll back to prev version for now

File: frontend/src/components/CollabPage/VideoChatPanel.jsx [hunk 1]
```diff
@@ -119,7 +119,8 @@ export default function VideoChatPanel() {
 
   const peer = new Peer({
     host: VIDEO_SVC_HOST,
-    port: VIDEO_SVC_PORT,
+    secure: true,
+    port: 443,
```
Comment (izzahaj) [inline frontend/src/components/CollabPage/VideoChatPanel.jsx]: Why does the port have to be 443?

Comment (wweqg) [reply frontend/src/components/CollabPage/VideoChatPanel.jsx]: Frontend makes https request now, so port 443 here is to ensure https connection can be made between the frontend and server. I will go and try some other configurations.

File: gateway/index.js [hunk 2]
```diff
@@ -15,7 +19,7 @@ const isAuthenticated = require('./authMiddleware');
 
 // Configure CORS for requests from anywhere
 const corsOptions = {
-  origin: 'http://localhost:3000',
+  origin: '*', // TODO: Update accordingly
```
Comment (izzahaj) [inline gateway/index.js:22]: Does this matter?

**Review by izzahaj — APPROVED** [review]
Works as per normal locally. On peerpreparing, profile page flashes a bit, but at this point i think it is not as crucial.

Also, do me a favour and delete the `/frontend/public/images` folder as we are no longer using it. it should stop the user-avatar errors in the browser console.

File: gateway/index.js (hunk 2 above)
Comment (wweqg) [reply gateway/index.js:22]: For actual production, it matters for security consideration. But I think for us its fine to just use "allow all".

**Pull request merged**